lib/matplotlib/axes/_base.py

```diff
@@ -2379,7 +2369,7 @@ def draw(self, renderer=None, inframe=False):
                 self.patch.get_path(),
                 self.patch.get_transform()))
 
-            renderer.draw_image(gc, round(l), round(b), im)
+            renderer.draw_image(gc, round(l), round(b), data)
```

Comment:
maybe draw it at the average zorder?  This is the time to break that sort of thing.